xcm/xcm-builder/src/tests/pay/pay.rs

```diff
@@ -0,0 +1,130 @@
+	pub InteriorBody: InteriorMultiLocation = Plurality { id: BodyId::Treasury, part: BodyPart::Voice }.into();
+	pub Timeout: BlockNumber = 5; // 5 blocks
+	pub Beneficiary: AccountId = 5;
+}
+
+/// Scenario:
+/// Account #3 on the local chain, Parachain(42), controls an amount of funds on Parachain(2).
+/// `PayOverXcm::pay` creates the correct message for account #3 to pay another account, account #5, on Parachain(2), remotely, in its native token.
+#[test]
+fn pay_over_xcm_works() {
+	let who = 5u64;
+	let asset_kind =
+		AssetKind { destination: (Parent, Parachain(2)).into(), asset_id: Here.into() };
+	let amount = 1000u128;
+
+	assert_ok!(PayOverXcm::<
+		InteriorAccount,
+		TestMessageSender,
+		TestQueryHandler<TestConfig, BlockNumber>,
+		Timeout,
+		AccountId,
+		AssetKind,
+		LocatableAssetKindConverter,
+		AliasesIntoAccountIndex64,
+	>::pay(&who, asset_kind, amount));
+
+	let expected_message = Xcm(vec![
+		DescendOrigin(AccountIndex64 { index: 3, network: None }.into()),
+		UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+		SetAppendix(Xcm(vec![ReportError(QueryResponseInfo {
+			destination: (Parent, Parachain(42)).into(),
+			query_id: 1,
+			max_weight: Weight::zero(),
+		})])),
+		TransferAsset {
+			assets: (Here, 1000u128).into(),
+			beneficiary: AccountIndex64 { index: 5, network: None }.into(),
+		},
+	]);
+
+	let expected_hash = fake_message_hash(&expected_message);
+	assert_eq!(sent_xcm(), vec![((Parent, Parachain(2)).into(), expected_message, expected_hash)]);
+}
+
+/// Scenario:
+/// A pluralistic body, a Treasury, on the local chain, Parachain(42), controls an amount of funds on Parachain(2).
+/// `PayOverXcm::pay` creates the correct message for the treasury to pay another account, account #7, on Parachain(2), remotely, in the relay's token.
+#[test]
+fn pay_over_xcm_governance_body() {
+	let who = 7u64;
+	let asset_kind =
+		AssetKind { destination: (Parent, Parachain(2)).into(), asset_id: Parent.into() };
+	let amount = 500u128;
+
+	assert_ok!(PayOverXcm::<
+		InteriorBody,
+		TestMessageSender,
+		TestQueryHandler<TestConfig, BlockNumber>,
+		Timeout,
+		AccountId,
+		AssetKind,
+		LocatableAssetKindConverter,
+		AliasesIntoAccountIndex64,
+	>::pay(&who, asset_kind, amount));
+
+	let expected_message = Xcm(vec![
+		DescendOrigin(Plurality { id: BodyId::Treasury, part: BodyPart::Voice }.into()),
+		UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+		SetAppendix(Xcm(vec![ReportError(QueryResponseInfo {
+			destination: (Parent, Parachain(42)).into(),
+			query_id: 1,
+			max_weight: Weight::zero(),
+		})])),
+		TransferAsset {
+			assets: (Parent, 500u128).into(),
+			beneficiary: AccountIndex64 { index: 7, network: None }.into(),
+		},
+	]);
+	let expected_hash = fake_message_hash(&expected_message);
+	assert_eq!(sent_xcm(), vec![((Parent, Parachain(2)).into(), expected_message, expected_hash)]);
```

Comment:
Similarly, an assertion of the transferred balance exists on account 7 would help.